distributed/tests/test_cancelled_state.py

```diff
@@ -140,20 +138,18 @@ async def test_worker_stream_died_during_comm(c, s, a, b):
     assert any("receive-dep-failed" in msg for msg in b.log)
 
 
-@gen_cluster(client=True)
-async def test_flight_to_executing_via_cancelled_resumed(c, s, a, b):
+@gen_cluster(client=True, nthreads=[("", 1)])
+async def test_flight_to_executing_via_cancelled_resumed(c, s, b):
+
     lock = asyncio.Lock()
     await lock.acquire()
 
-    async def wait_and_raise(*args, **kwargs):
-        async with lock:
-            raise CommClosedError()
+    class BrokenWorker(Worker):
+        async def get_data(self, comm, *args, **kwargs):
+            async with lock:
+                comm.abort()
```

Comment:
Woo!  +1 on using subclasses over mocks.  FWIW I find them to be both more understandable and less magical.